scribus/pageitem_textframe.cpp

```diff
@@ -2178,6 +2178,12 @@ void PageItem_TextFrame::layout()
 			// remember possible break
 			if (i + 1 < glyphRuns.count() && glyphRuns[i + 1].hasFlag(ScLayout_LineBoundry))
 			{
+
+				if( current.glyphs.length() > 1 && (current.glyphs[currentIndex -1].lastChar() != SpecialChars::CJK_NOBREAK_AFTER) &&
```

Comment:
The space should be before the opening bracket, also there should be spaces around both sides of the minus operator.